ARTEMIS-1639 HornetQClientProtocolManager sending unsupported packet

gaohoward opened the pull request:
HornetQClientProtocolManager is used to connect HornteQ servers.
During reconnect, it sends a CheckFailoverMessage packet to the
server as part of reconnection. This packet is not supported by
HornetQ server (existing release), so it will break the backward
compatibility.

clebertsuconic commented on tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/remoting/compat/HQClientProtocolManagerTest.java:
```diff
@@ -0,0 +1,104 @@
+import java.util.List;
+import java.util.Set;
+
+public class HQClientProtocolManagerTest extends ActiveMQTestBase {
```
Can you write a test using the new compatibility test framework instead?

it would be nice to do a simple failover of hornetq client, artemis 1.x and artemis 2.x to see if there's any further issues.

gaohoward replied:
ok will do.

gaohoward commented:
@clebertsuconic Hi Clebert, I added a new test to test the Artemis client failover on HornetQ server.
The set up is a live and a backup HornetQ server. Stop the live and the client fails over. 
Can you take a look?
Thanks

gaohoward commented:
@clebertsuconic forgot to mention that I fixed an issue where the connector configure passed from HornetQ server contains the hornetq's netty factory class name, which is not available in Artemis, I added a 'translation' to convert it to Artemis netty factory class.

clebertsuconic commented:
The test you added is broken...

clebertsuconic commented:
not running at all...always fail...

The compatibility testsuite is part of the fast-tests.. and gitpr check.. we cannot even add intermittent failures on this testsuite.. it has to be fast, reliable and reproduceable.

clebertsuconic commented on tests/compatibility-tests/src/main/resources/hqclientProtocolTest/failoverTest.groovy:
```diff
@@ -0,0 +1,85 @@
+
+    System.out.println("Message sent");
+
+    return connection;
```
if you are just returning the connection.. and using it on the test.. why are you using groovy here at all?

these groovy tests were meant to cross isolation of classLoaders. if you were mixing versions the return wouldn't work at all as the producer and consumer would be on different versions.

clebertsuconic commented on tests/compatibility-tests/src/main/resources/hqclientProtocolTest/failoverTest.groovy:
```diff
@@ -0,0 +1,85 @@
+    CompletionListener completionListener = new CompletionListener() {
+        @Override
+        void onCompletion(Message message) {
+            latch.countDown();
```
why are you using completion listener?

clebertsuconic commented:
There's no way how this would work.. you are using shared storage.. and each server on its own folder.

clebertsuconic commented:
I'm fixing it.

clebertsuconic commented:
Howard I made a few changes here. Please review.  It’s merged already.

gaohoward commented:
@clebertsuconic I've reviewed your changes already. It's good, thanks!
